volatility3/framework/layers/intel.py

```diff
@@ -307,6 +323,47 @@ class Intel32e(Intel):
 
 
 class WindowsMixin(Intel):
+    _swap_bit_offset = 32
+    def __init__(self,
+                 context: interfaces.context.ContextInterface,
+                 config_path: str,
+                 name: str,
+                 metadata: Optional[Dict[str, Any]] = None) -> None:
+        super().__init__(context = context, config_path = config_path, name = name, metadata = metadata)
+        self._kernel: Optional[interfaces.context.ModuleInterface] = self._get_kernel_module()
+
+    def _get_kernel_module(self) -> Optional[interfaces.context.ModuleInterface]:
+        kvo = self.config.get('kernel_virtual_offset', None)
+        if kvo is None:
+            return None
+
+        for module_name in self.context.modules:
+            if self.context.modules[module_name].offset == kvo:
+                return self.context.modules[module_name]
+
+        return None
+
+    @functools.lru_cache()
+    def _get_invalid_pte_mask(self, kernel):
+        if kernel.has_symbol("MiInvalidPteMask"):
+            pte_size = kernel.get_type("_MMPTE_HARDWARE").vol.size
+            pte_type = "unsigned int"
+            if pte_size == 8:
+                pte_type = "unsigned long long"
+
+            return kernel.object(pte_type, offset=kernel.get_symbol("MiInvalidPteMask").address)
+
+        if kernel.has_symbol("MiState"):
+            system_information = kernel.object("_MI_SYSTEM_INFORMATION", offset=kernel.get_symbol("MiState").address)
+            return system_information.Hardware.InvalidPteMask
+
+        return 0
```

Comment:
Is returning 0 the intended value?

Reply:
Yes, returning 0 causes the `xor` (`& ~`) operation later to just do nothing which is ok.